core-web/libs/block-editor/src/lib/extensions/image-tabview-form/image-tabview-form.component.ts

```diff
@@ -0,0 +1,161 @@
+import {
+    Component,
+    OnInit,
+    Input,
+    Output,
+    EventEmitter,
+    ViewChild,
+    ElementRef
+} from '@angular/core';
+import { FormBuilder, FormGroup } from '@angular/forms';
+import { BehaviorSubject } from 'rxjs';
+import { debounceTime, take, map, throttleTime, switchMap, filter } from 'rxjs/operators';
+import { Observable } from 'rxjs/internal/Observable';
+
+import { DotCMSContentlet } from '@dotcms/dotcms-models';
+
+import { DotLanguageService, Languages } from '../../shared';
+import {
+    ESOrderDirection,
+    queryEsParams,
+    SearchService
+} from '../../shared/services/search/search.service';
+
+@Component({
+    selector: 'dot-image-tabview-form',
+    templateUrl: './image-tabview-form.component.html',
+    styleUrls: ['./image-tabview-form.component.scss']
+})
+export class ImageTabviewFormComponent implements OnInit {
+    @Input() languageId = 1;
+    @Output() selectedContentlet: EventEmitter<DotCMSContentlet> = new EventEmitter();
+    @ViewChild('inputSearch') inputSearch!: ElementRef;
+
+    loading = true;
+    done = false;
+    search$ = new BehaviorSubject<number>(null);
+    contentlets$: Observable<DotCMSContentlet[][]>;
+    virtualItems: DotCMSContentlet[][] = [];
+    form: FormGroup;
+
+    private dotLangs: Languages;
+
+    get currentSearch() {
+        return this.form.get('search').value;
+    }
+
+    constructor(
+        private searchService: SearchService,
+        private dotLanguageService: DotLanguageService,
+        private fb: FormBuilder
+    ) {}
+
+    ngOnInit(): void {
+        this.form = this.fb.group({
+            search: ['']
+        });
+
+        this.contentlets$ = this.search$.pipe(
+            filter((x) => x != null),
+            throttleTime(500),
+            // offset * 2 -> Because of the virtual scroll ([[DotCMSContentlet, DotCMSContentlet], ...])
+            switchMap((offset) => this.searchContentlets(Math.ceil(offset * 2)))
+        );
```

Comment:
```suggestion
        this.contentlets$ = fromEvent(input, 'change').pipe(
            filter((x) => x != null),
            throttleTime(500),
            // offset * 2 -> Because of the virtual scroll ([[DotCMSContentlet, DotCMSContentlet], ...])
            switchMap((offset) => this.searchContentlets(Math.ceil(offset * 2)))
        );
```

Reply:
And this is cleaner cuz' you don't need the form.